kuma/wiki/tests/test_models_document.py

```diff
@@ -43,6 +43,42 @@ def doc_with_sections(root_doc, wiki_user):
     return root_doc
 
 
+@pytest.mark.parametrize('locales,expected_results',
+                         HREFLANG_TEST_CASES.values(),
+                         ids=HREFLANG_TEST_CASES.keys())
+def test_document_get_hreflang(root_doc, locales, expected_results):
+    docs = [
+        Document.objects.create(
+            locale=locale,
+            slug='Root',
+            title='Root Document',
+            rendered_html='<p>...</p>',
+            parent=root_doc
+        ) for locale in locales
+    ]
+    for doc, expected_result in zip(docs, expected_results):
+        assert doc.get_hreflang() == expected_result
+
+
+@pytest.mark.parametrize('locales,expected_results',
+                         HREFLANG_TEST_CASES.values(),
+                         ids=HREFLANG_TEST_CASES.keys())
+def test_document_get_hreflang_with_other_locales(root_doc, locales,
+                                                  expected_results):
+    for locale, expected_result in zip(locales, expected_results):
+        doc = Document.objects.create(
+            locale=locale,
+            slug='Root',
+            title='Root Document',
+            rendered_html='<p>...</p>',
+            parent=root_doc
+        )
+        with mock.patch.object(doc, 'get_other_translations',
+                               return_value=()) as mocked_method:
```

Comment:
When I'm mocking to ensure a method isn't called, I use ``side_effect=Exception('with optional string')``, so that an exception would be raised if the method was called. With a ``return_value``, it implies you are expecting a call, and returning an empty sequence for test purposes.

Reply:
I like that much better, thanks @jwhitlock! I will update this PR with that change and then merge.